storagemarket/provider_test.go

```diff
@@ -686,6 +682,9 @@ func NewHarness(t *testing.T, ctx context.Context, opts ...harnessOpt) *Provider
 
 	ph.MockSealingPipelineAPI.EXPECT().SectorsSummary(gomock.Any()).Return(sealingpipelineStatus, nil).AnyTimes()
 
```

Comment:
Please can we add verification for getting an event for atleast one of the sealing states here ?

Reply:
When the deal has been added to a piece, the deal go routine starts polling the sealer for sealing state updates. I've modified the sealing API mock to return the "Proving" state. When the code polling the sealer gets sealing state "Proving" it stops polling and returns. So given that the tests finish correctly, that verifies that the event code path is working correctly.

Let me know if you think we need other tests beyond this one.

Reply:
But shouldn't we see an event for that sealing state update to `Proving` ?

Reply:
I added a test to watch for an event to be fired with sealing state `Proving`